doc/smart-switch/ip-address-assigment/smart-switch-ip-address-assignment.md

```diff
@@ -199,53 +222,123 @@ The DHCP_SERVER_IPV4_PORT table includes the following:
-- (2) sonic-cfggen pushes configuration into the Config DB.
-- (3-4) DHCP server and DHCP relay containers upon start consume configuration from the config DB and start listening for a requests from DHCP clients.
-- (5) DHCP client on the DPU sends a request over the eth0 interfaces. The request comes to the DHCP relay through the midplane bridge. DHCP relay inserts option 82 into the request with the information about the interface from which the request came. DHCP relay forwards the packet to the DHCP server.
-- (6) The DHCP server sends a reply with the IP configuration to the DHCP client on the DPU.
+``` text
+# database.service
+
+[Unit]
+...
+Requires=midplane-network.service
+After=midplane-network.service
+
+...
+```
+
+2. systemd-networkd helps to create the bridge-midplane interface and assign the specific IP address according to above configuration. Meanwhile, systemd-networkd will monitor the DPU PCIe interface. Once the PCIe interface is created, it will automatically add it into the bridge-midplane.
+3. midplane-network.service will be used to wait for the midplane bridge configured.
 
-### DPU ###
+``` text
+# midplane-network.service
 
-In the Smart Switch DPU, the PCIe interface should be used as a default management interface for communication with the switch and the outside world.
+[Unit]
+Description=Midplane network service
+Requires=systemd-networkd.service
+After=systemd-networkd.service
 
-It is the vendors' responsibility to make sure that the PCIe interface that is connected to the switch has given the `eth0` name.
+[Service]
+Type=oneshot
+User=root
+ExecStart=/usr/lib/systemd/systemd-networkd-wait-online -i bridge-midplane
 
-The IP address assignment for the default `eth0` management interface in the existing SONiC implementation is the following:
+[Install]
+WantedBy=multi-user.target
+```
+
+``` text
+# database.service
+
+[Unit]
+...
+Requires=midplane-network.service
+After=midplane-network.service
+
+...
+```
+
+#### DPU ####
+
+In the DPU side, the steps are almost similar to NPU. But the generated configuration is different but simpler.
+
+``` text
+# midplane-network.network
+[Match]
+Name=eth0-midplane
+
+[Network]
+DHCP=yes
+```
+
+``` text
+# midplane-network.service
 
-- If the `MGMT_INTERFACE` table from Config DB is empty
-  - Query the IP address for the `eth0` interface from the DHCP server
-- Else
-  - For each entry
-    - Assign static IP configuration for the interface
+[Unit]
+Description=Midplane network service
+Requires=systemd-networkd.service
+After=systemd-networkd.service
 
-To trigger the DPU SONiC to query the IP address configuration from the DHCP server the `MGMT_INTERFACE` in the config DB MUST remain empty. This also ensures that the DPU SONiC image can boot and bring the management interface up automatically with the default configuration.
+[Service]
+Type=oneshot
+User=root
+ExecStart=/usr/lib/systemd/systemd-networkd-wait-online -i eth0-midplane --timeout=600
```

Comment:
@Pterosaur systemd-networkd already knows the interfaces it is managing and hence will wait for a default timeout for the interface to be ready. Why this explicit timeout for DPU but NOT for NPU case where you are not specifying similar wait for dpu0, dpu1, ..interfaces?

Reply:
In the NPU side, we don't mind when these DPU PCIe interfaces are created/added into the midplane bridge. So, in the NPU side, we just need to wait for the bridge-midplane.